src/apps/statements/service/storeStatements/index.ts

```diff
@@ -33,20 +34,25 @@ const cloneAttachments = (attachmentModels: AttachmentModel[]): AttachmentModel[
 };
 
 export default (config: Config) => {
-  return async (opts: StoreStatementsOptions): Promise<string[]> => {
-    checkScopes(STATEMENT_WRITE_SCOPES, opts.client.scopes);
-    const preValidatedModels = preValidationSetup(config, opts.models);
+  return async ({
+    models,
+    attachments,
+    client,
+    priority = StatementProcessingPriority.MEDIUM,
+  }: StoreStatementsOptions): Promise<string[]> => {
+    checkScopes(STATEMENT_WRITE_SCOPES, client.scopes);
+    const preValidatedModels = preValidationSetup(config, models);
     validateStatements(preValidatedModels);
-    const attachments = cloneAttachments(opts.attachments);
-    const clonedAttachments = cloneAttachments(opts.attachments);
+    const clonedAttachments = cloneAttachments(attachments);
     const postValidatedModels = await postValidationSetup(
       preValidatedModels,
-      clonedAttachments,
-      opts.client,
+      cloneAttachments(attachments),
+      client,
+      priority,
     );
-    const unstoredModels = await getUnstoredModels(config, postValidatedModels, opts.client);
-    const voidedObjectIds = await checkVoiders(config, unstoredModels, opts.client);
-    checkAttachments(config, postValidatedModels, attachments);
+    const unstoredModels = await getUnstoredModels(config, postValidatedModels, client);
+    const voidedObjectIds = await checkVoiders(config, unstoredModels, client);
+    checkAttachments(config, postValidatedModels, clonedAttachments);
```

Comment:
I'm assuming this is correct, looks very intentional 👍  I feel like we might have even discussed this at some point.